File: puzzles/python3/ascii-art/ascii_art.py
if __name__ == "__main__":
    width: int = int(input())
    height: int = int(input())
    text: str = input().upper()
    for _ in range(height):
        row: str = input()
        output: str = ""
        for character in text:
            position: int = ord(character) - ord('A')
            if position < 0 or position > 25:
                position = 26
            start: int = position * width
            end: int = start + width
            output += row[start:end]
        print(output)

File: puzzles/perl/ascii-art.pl
use strict;
use warnings;
use 5.20.1;

select(STDOUT); $| = 1; # DO NOT REMOVE

chomp(my $width = <STDIN>);
chomp(my $height = <STDIN>);
chomp(my $text = <STDIN>);
for my $i (0..$height-1) {
  chomp(my $row = <STDIN>);
  my $output = "";
  for my $j (0..(length($text)-1)) {
    my $character = uc substr($text, $j, 1);
    my $position = ord($character) - ord('A');
    if ($position < 0 || $position > 25) {
      $position = 26;
    }
    my $start = $position * $width;
    $output .= substr($row, $start, $width);
  }
  print "$output\n";
}
